test/unit/strategies/RWAStrategy.js

```diff
@@ -0,0 +1,400 @@
+            // Check that the funds were transferred to the manager
+            const managerBalance = await asset.balanceOf(manager.address);
+            expect(managerBalance).to.equal(deployAmount);
+    
+            // Verify totalInvestedInRWA was updated correctly
+            const rwaStrategy = await ethers.getContractAt("RWAStrategy", strategy.target);
+            const totalInvestedInRWA = await rwaStrategy.totalInvestedInRWA();
+            expect(totalInvestedInRWA).to.equal(deployAmount);
+        });
+    
+        it("Does not transfer funds if amount does not exceed minAmountToSell", async function () {
+            const { vault, strategy, asset, mockAsset, deployer, manager } = await loadFixture(deployRWAStrategyFixture);
+    
+            const deployAmount = 50;
+            await asset.approve(vault.target, deployAmount);
+            await vault.setDepositLimit(deployAmount);
+            await vault.deposit(deployAmount, deployer.address);
+    
+            // Record the initial manager balance
+            const initialManagerBalance = await asset.balanceOf(manager.address);
+    
+            // Attempt to deploy funds
+            await vault.updateDebt(strategy.target, deployAmount);
+    
+            // Check that the funds were not transferred
+            const managerBalance = await asset.balanceOf(manager.address);
+            expect(managerBalance).to.equal(initialManagerBalance);
+    
+            // Verify totalInvestedInRWA was not updated
+            const rwaStrategy = await ethers.getContractAt("RWAStrategy", strategy.target);
+            const totalInvestedInRWA = await rwaStrategy.totalInvestedInRWA();
+            expect(totalInvestedInRWA).to.equal(0); // Assuming this starts at 0
+        });
+    });
+
+    describe("_freeFunds()", function () {
+        it("Successfully transfers funds and updates totalInvestedInRWA on withdrawal", async function () {
+            const { vault, strategy, asset, manager, deployer, otherAccount } = await loadFixture(deployRWAStrategyFixture);
+            
+            // Simulate the strategy investing funds
+            const deployAmount = 2000;
+            await asset.approve(vault.target, deployAmount);
+            await vault.setDepositLimit(deployAmount);
+            await vault.deposit(deployAmount, deployer.address);
+
+            await expect(vault.updateMaxDebtForStrategy(strategy.target, deployAmount))
+                .to.emit(vault, 'UpdatedMaxDebtForStrategy')
+                .withArgs(deployer.address, strategy.target, deployAmount);
+            
+            await vault.updateDebt(strategy.target, deployAmount);
+
+            // Approve the strategy to spend the asset on behalf of the manager
+            await asset.connect(manager).approve(strategy.target, deployAmount);
+    
+            // Perform withdrawal through the vault, triggering _freeFunds
+            const withdrawalAmount = 50;
+            await vault.withdraw(
```

Comment:
So basically when user wants the funds back - he can force manager to return them if manager has approved it?

Reply:
Yes, and if manager has enough balance of course, because if manager has bought all in RWAs this will simply fail by design.